OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -261,18 +262,24 @@ void SanityCheckPath(Mobile mobile)
 			return Pair.New(nextCell, subCell);
 		}
 
+		public override Color TargetLineColor
+		{
+			get
+			{
+				if (NextActivity != null)
+					return NextActivity.TargetLineColor;
+				return Color.Green;
+			}
+		}
+
 		public override IEnumerable<Target> GetTargets(Actor self)
 		{
-			if (path != null)
-				return Enumerable.Reverse(path).Select(c => Target.FromCell(self.World, c));
-			if (destination != null)
-				return new Target[] { Target.FromCell(self.World, destination.Value) };
-			return Target.None;
+			yield return Target.FromCell(self.World, this.destination.Value);
 		}
 
-		abstract class MovePart : Activity
+		public abstract class MovePart : Activity
```

Comment:
In relation to below struct approach, this gives rise to make struct a linked list, as this MovePart activity is quite different from other activities. It doesn't have NextActivity nor ParentActivity but in practice, the member variable Move is the next activity. Then the below struct must become a class.